src/mode.ts

```diff
@@ -30,16 +48,23 @@ export function setupJavaScript(defaults:LanguageServiceDefaultsImpl): void {
 	);
 }
 
-function setupMode(defaults:LanguageServiceDefaultsImpl, modeId:string, language:Language): void {
+export function getEmitOutput(uri: monaco.Uri): monaco.Promise<any> {
+	return executeAfterSetup((uri: monaco.Uri) => {
+		return worker(uri)
+				.then((worker: TypeScriptWorker) =>
+					worker.getEmitOutput(uri.toString()))
+		}, null, [uri]);
+}
 
+function setupMode(defaults:LanguageServiceDefaultsImpl, modeId:string, language:Language): void {
 	let disposables: IDisposable[] = [];
 
-	const client = new WorkerManager(defaults);
-	disposables.push(client);
+	if (setupModes.indexOf(modeId) > -1) {
+		return;
+	}
 
-	const worker = (first: Uri, ...more: Uri[]): Promise<TypeScriptWorker> => {
-		return client.getLanguageServiceWorker(...[first].concat(more));
-	};
+	client = new WorkerManager(defaults);

```

Comment:
No, both are two separate 'extensions' and cannot be married in that way cos it will disallow having different configurations (compiler options, extra libs) for each. That doesn't mean you cannot have JS files in TypeScript - use the `allowJS`-options and add them as `extra lib` or just add JS content as TS-file.